calyx-py/calyx/builder.py

```diff
@@ -118,6 +118,14 @@ def input(self, name: str, size: int) -> ExprBuilder:
         self.component.inputs.append(ast.PortDef(ast.CompVar(name), size))
         return self.this()[name]
 
+    
+    def input_with_attributes(self, name: str, size: int, attribute_literals: List[Union[str, Tuple[str, int]]]) -> ExprBuilder:
```

Comment:
Just a UX thing, but consider making the list of attributes an optional argument in the existing `input` method? The bodies of the two methods are quite different, I realize, but it may be nice for the user to have a unified experience instead of having to use `input_with_attributes`!

Same with output...

Reply:
Wow, I had almost exactly the same comment at almost exactly the same instant!

Reply:
I debated this a bit myself and settled on having a separate method because in the past I've been told it's better "tell not ask", so having the user explicitly call a separate method when attributes are wanted was based on that advice. I will change this, and it's not a big deal I think, but am curious if there are thoughts about this more generally

Reply:
That's an interesting take! I don't have a strong opinion and lack any theoretical underpinning haha, but I guess in this case it "just" feels like "an input port,  but with more razzle dazzle" and so it kinda feels like one thing?